chalk-solve/src/wf.rs

```diff
@@ -609,7 +617,7 @@ impl WellKnownConstraints {
             ..
         }) = impl_datum
             .binders
-            .value
+            .skip_binders()
             .trait_ref
             .self_type_parameter(interner)
```

Comment:
Side note that I think we should add some accessor here to `ImplDatum` that encapsulates this `skip_binders` call. Something like

```rust
/// If the self-type for this impl is a struct, returns its id.
fn self_type_struct_id(&self) -> Option<StructId>
```

I think it's probably useful in other places (e.g., the auto-trait code potentially?) and regardless I generally prefer to have `skip_binders` logic encapsulated so it be more easily reviewed. But maybe we should add it as a separate "cleanup item", because I think there are other `skip_binders` calls scattered about that could use similar treatment.

Reply:
I agree, we have a number of "get trait id" and "get struct id" uses of `skip_binders()` that can be moved to helper methods